src/persist-client/src/internal/maintenance.rs

```diff
@@ -196,38 +207,47 @@ where
         V: Debug + Codec,
         D: Semigroup + Codec64 + Send + Sync,
     {
-        for future in self.perform_in_background(machine, gc, compactor) {
-            let _ = future.await;
-        }
+        self.perform_in_background(machine, gc, compactor).await
     }
 
     /// Initiates maintenance work in the background, either through spawned tasks
     /// or by sending messages to existing tasks. The returned futures may be
     /// awaited to know when the work is completed, but do not need to be polled
     /// to drive the work to completion.
```

Comment:
can update this comment since it no longer returns futures

Reply:
Huh, yeah, looking at it now I don't know there's much advantage to having both `perform` and `perform_in_background` here. I'll collapse them and see what you think!

Reply:
Pushed in a [separate commit](https://github.com/MaterializeInc/materialize/pull/17717/commits/04e59d34f3c7f2b4d6613bb27813350207a6cb62)!